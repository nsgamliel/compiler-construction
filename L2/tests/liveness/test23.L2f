  (:go
0

    // Use r12
    %variableK1MKK5A9B6 <- r12
    r12 <- 1

    // Print r12
    rdi <- r12
    call print 1

    %var1 <- 5
    mem rsp -8 <- :myF_ret
    call :myF 0
    :myF_ret

    %var1++

    // Restore r12
    r12 <- %variableK1MKK5A9B6

    return
  )
