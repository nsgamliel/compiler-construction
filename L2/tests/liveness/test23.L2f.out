(
(in
(r12 r13 r14 r15 rbp rbx)
(r13 r14 r15 rbp rbx %variableK1MKK5A9B6)
(r12 r13 r14 r15 rbp rbx %variableK1MKK5A9B6)
(r13 r14 r15 rbp rbx rdi %variableK1MKK5A9B6)
(r13 r14 r15 rbp rbx %variableK1MKK5A9B6)
(r13 r14 r15 rbp rbx %var1 %variableK1MKK5A9B6)
(r13 r14 r15 rbp rbx %var1 %variableK1MKK5A9B6)
(r13 r14 r15 rax rbp rbx %var1 %variableK1MKK5A9B6)
(r13 r14 r15 rax rbp rbx %var1 %variableK1MKK5A9B6)
(r13 r14 r15 rax rbp rbx %variableK1MKK5A9B6)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(r13 r14 r15 rbp rbx %variableK1MKK5A9B6)
(r12 r13 r14 r15 rbp rbx %variableK1MKK5A9B6)
(r13 r14 r15 rbp rbx rdi %variableK1MKK5A9B6)
(r13 r14 r15 rbp rbx %variableK1MKK5A9B6)
(r13 r14 r15 rbp rbx %var1 %variableK1MKK5A9B6)
(r13 r14 r15 rbp rbx %var1 %variableK1MKK5A9B6)
(r13 r14 r15 rax rbp rbx %var1 %variableK1MKK5A9B6)
(r13 r14 r15 rax rbp rbx %var1 %variableK1MKK5A9B6)
(r13 r14 r15 rax rbp rbx %variableK1MKK5A9B6)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
